packages/bundler-plugin-core/src/plugins/react-annotate-plugin.js

```diff
@@ -0,0 +1,480 @@
+/**
+ * This plugin is vendored in from FullStory's Babel React Annotate plugin:
+ * https://github.com/fullstorydev/fullstory-babel-plugin-annotate-react/tree/master
+ */
+
+const webComponentName = "data-sentry-component";
+const webElementName = "data-sentry-element";
+const webSourceFileName = "data-sentry-source-file";
+
+const nativeComponentName = "dataSentryComponent";
+const nativeElementName = "dataSentryElement";
+const nativeSourceFileName = "dataSentrySourceFile";
+const fsTagName = "fsTagName";
+
+const annotateFragmentsOptionName = "annotate-fragments";
+const excludedComponentsOptionName = "excludedComponents";
+
+const knownIncompatiblePlugins = [
+  // This module might be causing an issue preventing clicks. For safety, we won't run on this module.
+  "react-native-testfairy",
+  // This module checks for unexpected property keys and throws an exception.
+  "@react-navigation",
+];
+
+export default function reactAnnotate({ types: t }) {
+  return {
+    pre() {
+      this.excludedComponents = this.opts[excludedComponentsOptionName] || [];
+      if (this.opts.setFSTagName && !this.opts.native) {
+        throw new Error(
+          "`setFSTagName: true` is invalid unless `native: true` is also set in the configuration for @fullstory/babel-plugin-annotate-react"
+        );
+      }
+    },
+    visitor: {
+      FunctionDeclaration(path, state) {
+        if (!path.node.id || !path.node.id.name) return;
```

Comment:
l: let's do full return statements here to help with readability

```js
if (!path.node.id || !path.node.id.name) {
  return;
}
```

Ditto with everything else.